src/plugins/intel_cpu/src/emitters/snippets/cpu_runtime_configurator.cpp

```diff
@@ -4,30 +4,57 @@
 
 #include "emitters/snippets/cpu_runtime_configurator.hpp"
 
-#include "snippets/utils/utils.hpp"
 #include "snippets/lowered/loop_manager.hpp"
-
+#include "snippets/lowered/pass/init_loops.hpp"
+#include "snippets/pass/split_dimension_m.hpp"
+#include "snippets/utils/utils.hpp"
 
 namespace ov {
 namespace intel_cpu {
 
+using namespace ov::snippets::lowered;
+using namespace ov::snippets::pass;
+
+const size_t CPURuntimeConfigurator::rank6D = 6;
+
 CPURuntimeConfigurator::CPURuntimeConfigurator() : ov::snippets::RuntimeConfigurator(std::make_shared<CPURuntimeConfig>()) {
 }
 
 void CPURuntimeConfigurator::update(const ov::snippets::lowered::LinearIRPtr& linear_ir) {
+    m_config->master_shape = linear_ir->get_master_shape();
+
+    ov::snippets::RuntimeConfigurator::LoopInfoRuntimeParamsMap initialized_info;
+    std::vector<ov::snippets::VectorDims> updated_shapes;
+    std::vector<std::vector<size_t>> updated_layouts;
+    bool optimize_work_amount = m_optimizer.need_optimize(m_config->master_shape);
+    if (optimize_work_amount) {
+        m_optimizer.update_split_loops_info(initialized_info);
+        m_optimizer.update_shapes(m_io_descs, updated_shapes, m_in_num);
+        m_optimizer.update_layouts(m_io_descs, updated_layouts, m_in_num);
+        m_optimizer.update_config(m_config);
+    }
+
     if (linear_ir->is_dynamic()) {
-        update_loop_info(linear_ir);
+        update_loop_info(linear_ir, initialized_info);
         update_loop_args(linear_ir);
         // Update KernelExecutor Table should be before `update_buffer_scratchpad_size`
         // because `ComputeAllocationSize` depends on subtensors which are updated in the table
         get_kernel_executor_table()->update_state(linear_ir);
         update_buffer_scratchpad_size(linear_ir);
     }
 
-    m_config->master_shape = linear_ir->get_master_shape();
+    update_data_offsets(updated_shapes, updated_layouts);
+    // TODO: unify this logic somehow?
+    if (!optimize_work_amount) {
+        update_latest_shapes();
+    } else {
+        m_latest_shapes = std::move(updated_shapes);
+    }
```

Comment:
To discuss: If we updated `m_latest_shapes` within `update_data_offsets`, we would not need to split this logic depending on `optimize_work_amount`.

Reply:
Not relevant anymore